lib/logstash/filters/mutate.rb

```diff
@@ -338,85 +334,91 @@ def gsub_dynamic_fields(event, original, needle, replacement)
     end
   end
 
-  private
   def uppercase(event)
     @uppercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.upcase! }
-      elsif event[field].is_a?(String)
-        event[field].upcase!
-      else
-        @logger.debug("Can't uppercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:upcase!)
+        when String
+          original.upcase!
+        else
+          @logger.debug("Can't uppercase something that isn't a string",
+                        :field => field, :value => original)
+          original
+        end
     end
   end # def uppercase
 
-  private
   def lowercase(event)
     @lowercase.each do |field|
-      if event[field].is_a?(Array)
-        event[field].each { |v| v.downcase! }
-      elsif event[field].is_a?(String)
-        event[field].downcase!
-      else
-        @logger.debug("Can't lowercase something that isn't a string",
-                      :field => field, :value => event[field])
-      end
+      original = event[field]
+      event[field] = case original
+        when Array
+          original.map(&:downcase!)
+        when String
+          original.downcase!
+        else
+          @logger.debug("Can't lowercase something that isn't a string",
+                        :field => field, :value => original)
+          original
+        end
     end
   end # def lowercase
 
-  private
   def split(event)
     @split.each do |field, separator|
-      if event[field].is_a?(String)
-        event[field] = event[field].split(separator)
+      original = event[field]
+      if original.is_a?(String)
+        event[field] = original.split(separator)
       else
         @logger.debug("Can't split something that isn't a string",
-                      :field => field, :value => event[field])
+                      :field => field, :value => original)
       end
     end
   end
 
-  private
   def join(event)
     @join.each do |field, separator|
-      if event[field].is_a?(Array)
-        event[field] = event[field].join(separator)
+      original = event[field]
+      if original.is_a?(Array)
+        event[field] = original.join(separator)
       end
     end
   end
 
-  private
   def strip(event)
     @strip.each do |field|
-      if event[field].is_a?(Array)
-        event[field] = event[field].map{|s| s.strip }
-      elsif event[field].is_a?(String)
-        event[field] = event[field].strip
+      original = event[field]
+      if original.is_a?(Array)
```

Comment:
I considered it.  It did not have an else branch, so I thought is might look as though the else case was omitted by mistake.
A better one that I missed is the convert method.  I should change that.